File: replace.py
import os
import re

def replace_in_dir(directory):
    for root, dirs, files in os.walk(directory):
        for file in files:
            if file.endswith(('.ts', '.tsx', '.css')):
                full_path = os.path.join(root, file)
                with open(full_path, 'r', encoding='utf-8') as f:
                    content = f.read()
                
                # Replace paths like from "../lib/api" or from "../../lib/api" or import "../../styles/tokens.css"
                content = re.sub(r'from\s+["\'](?:\.\./)+lib/api["\']', 'from "@ddup/shared/lib/api"', content)
                content = re.sub(r'from\s+["\'](?:\.\./)+contexts/displayMode["\']', 'from "@ddup/shared/contexts/displayMode"', content)
                content = re.sub(r'from\s+["\'](?:\.\./)+ui["\']', 'from "@ddup/shared/ui"', content)
                content = re.sub(r'from\s+["\'](?:\.\./)+ui/([^"\']+)["\']', r'from "@ddup/shared/ui/\1"', content)
                content = re.sub(r'import\s+["\'](?:\.\./)+styles/([^"\']+)["\']', r'import "@ddup/shared/styles/\1"', content)
                
                # Also handle ./ imports if they exist (e.g., in App.tsx they might be ./lib/api)
                content = re.sub(r'from\s+["\']\./lib/api["\']', 'from "@ddup/shared/lib/api"', content)
                content = re.sub(r'from\s+["\']\./contexts/displayMode["\']', 'from "@ddup/shared/contexts/displayMode"', content)
                content = re.sub(r'from\s+["\']\./ui["\']', 'from "@ddup/shared/ui"', content)
                content = re.sub(r'from\s+["\']\./ui/([^"\']+)["\']', r'from "@ddup/shared/ui/\1"', content)
                content = re.sub(r'import\s+["\']\./styles/([^"\']+)["\']', r'import "@ddup/shared/styles/\1"', content)
                
                with open(full_path, 'w', encoding='utf-8') as f:
                    f.write(content)

replace_in_dir('apps/pc/src')
replace_in_dir('apps/mobile/src')

File: replace.js
const fs = require('fs');
const path = require('path');

function replaceInDir(dir) {
  const files = fs.readdirSync(dir);
  for (const file of files) {
    const fullPath = path.join(dir, file);
    if (fs.statSync(fullPath).isDirectory()) {
      replaceInDir(fullPath);
    } else if (fullPath.endsWith('.ts') || fullPath.endsWith('.tsx') || fullPath.endsWith('.css')) {
      let content = fs.readFileSync(fullPath, 'utf8');
      const originalContent = content;
      
      content = content.replace(/from\s+["'](\.\.\/)+lib\/api["']/g, 'from "@ddup/shared/lib/api"');
      content = content.replace(/from\s+["'](\.\.\/)+contexts\/displayMode["']/g, 'from "@ddup/shared/contexts/displayMode"');
      content = content.replace(/from\s+["'](\.\.\/)+ui["']/g, 'from "@ddup/shared/ui"');
      content = content.replace(/from\s+["'](\.\.\/)+ui\/(.*?)["']/g, 'from "@ddup/shared/ui/$2"');
      content = content.replace(/import\s+["'](\.\.\/)+styles\/(.*?)["']/g, 'import "@ddup/shared/styles/$2"');
      
      content = content.replace(/from\s+["']\.\/lib\/api["']/g, 'from "@ddup/shared/lib/api"');
      content = content.replace(/from\s+["']\.\/contexts\/displayMode["']/g, 'from "@ddup/shared/contexts/displayMode"');
      content = content.replace(/from\s+["']\.\/ui["']/g, 'from "@ddup/shared/ui"');
      content = content.replace(/from\s+["']\.\/ui\/(.*?)["']/g, 'from "@ddup/shared/ui/$1"');
      content = content.replace(/import\s+["']\.\/styles\/(.*?)["']/g, 'import "@ddup/shared/styles/$1"');
      
      if (content !== originalContent) {
        fs.writeFileSync(fullPath, content);
        console.log(`Updated: ${fullPath}`);
      }
    }
  }
}

replaceInDir('apps/pc/src');
replaceInDir('apps/mobile/src');
console.log('Done!');
